yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,396 @@
+        amount = Math.min(amount, balanceOfPool());
+        uint256 initialWantBalance = balanceOfWant();
+        ILPRewards(tbtcSaddleLPRewards).withdraw(amount);
+        return balanceOfWant().sub(initialWantBalance);
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to liquidate up to amountNeeded of want token of this strategy's
+    ///         positions, irregardless of slippage. Any excess will be
+    ///         re-invested with adjustPosition. This function should return
+    ///         the amount of want tokens made available by the liquidation.
+    ///         If there is a difference between them, loss indicates whether
+    ///         the difference is due to a realized loss, or if there is some
+    ///         other situation at play (e.g. locked funds). This strategy
+    ///         implements the aforementioned behavior by withdrawing a portion
+    ///         of the vault's underlying token (want token) from the tBTC v2
+    ///         Saddle reward pool.
+    /// @dev The invariant `liquidatedAmount + loss <= amountNeeded` should
+    ///      always be maintained.
+    /// @param amountNeeded Amount of the vault's underlying tokens needed by
+    ///        the liquidation process.
+    /// @return liquidatedAmount Amount of vault's underlying tokens made
+    ///         available by the liquidation.
+    /// @return loss Amount of the loss.
+    function liquidatePosition(uint256 amountNeeded)
+        internal
+        override
+        returns (uint256 liquidatedAmount, uint256 loss)
+    {
+        uint256 wantBalance = balanceOfWant();
+        if (wantBalance < amountNeeded) {
+            liquidatedAmount = withdrawSome(amountNeeded.sub(wantBalance));
+            liquidatedAmount = liquidatedAmount.add(wantBalance);
+            loss = amountNeeded.sub(liquidatedAmount);
+        } else {
+            liquidatedAmount = amountNeeded;
+        }
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to liquidate everything and return the amount that got freed.
+    ///         This strategy implements the aforementioned behavior by
+    ///         withdrawing all vault's underlying tokens from the tBTC v2 Saddle
+    ///         reward pool.
+    /// @dev This function is used during emergency exit instead of prepareReturn
+    ///      to liquidate all of the strategy's positions back to the vault.
+    /// @return amountFreed Amount that got freed.
+    function liquidateAllPositions()
+        internal
+        override
+        returns (uint256 amountFreed)
+    {
+        ILPRewards(tbtcSaddleLPRewards).withdraw(balanceOfPool());
+
+        // Yearn docs doesn't specify clear enough what exactly should be
+        // returned here. It may be either the total balance after
+        // withdrawAllAndUnwrap or just the amount withdrawn. Currently opting
```

Comment:
- [... after `withdraw` or just ...] 
- And since we're returning the entire balance and not only Pool's rewards portion, then we should probably reword @return param to "Total balance of want tokens". "Amount that got freed" denotes what was freed from the `tbtcSaddleLPRewards` pool and not the entire balance.

Reply:
withdrawAllAndUnwrap -> withdraw

Reply:
Fixed in https://github.com/keep-network/tbtc-v2/pull/13/commits/34980e454124e22d92e144195e0a2062c4eb1b84.